src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1311,8 +1311,183 @@ public static Matrix4x4 CreateReflection(Plane value)
+                C4 = Sse.Subtract(C4, Sse.Multiply(V02, V12));
+                C6 = Sse.Subtract(C6, Sse.Multiply(V03, V13));
+                V00 = Sse.Shuffle(row2, row2, 0x33);
+
+                // V10 = D0Z,D0Z,D2X,D2Y
+                V10 = Sse.Shuffle(D0, D2, 0x4A);
+                V10 = Sse.Shuffle(V10, V10, 0x2C);
+                V01 = Sse.Shuffle(row1, row1, 0x8D);
+
+                // V11 = D0X,D0W,D2X,D2Y
+                V11 = Sse.Shuffle(D0, D2, 0x4C);
+                V11 = Sse.Shuffle(V11, V11, 0x93);
+                V02 = Sse.Shuffle(row4, row4, 0x33);
+
+                // V12 = D1Z,D1Z,D2Z,D2W
+                V12 = Sse.Shuffle(D1, D2, 0xEA);
+                V12 = Sse.Shuffle(V12, V12, 0x2C);
+                V03 = Sse.Shuffle(row3, row3, 0x8D);
+
+                // V13 = D1X,D1W,D2Z,D2W
+                V13 = Sse.Shuffle(D1, D2, 0xEC);
+                V13 = Sse.Shuffle(V13, V13, 0x93);
+
+                V00 = Sse.Multiply(V00, V10);
+                V01 = Sse.Multiply(V01, V11);
+                V02 = Sse.Multiply(V02, V12);
+                V03 = Sse.Multiply(V03, V13);
+
+                Vector128<float> C1 = Sse.Subtract(C0, V00);
+                C0 = Sse.Add(C0, V00);
+                Vector128<float> C3 = Sse.Add(C2, V01);
+                C2 = Sse.Subtract(C2, V01);
+                Vector128<float> C5 = Sse.Subtract(C4, V02);
+                C4 = Sse.Add(C4, V02);
+                Vector128<float> C7 = Sse.Add(C6, V03);
+                C6 = Sse.Subtract(C6, V03);
+
+                C0 = Sse.Shuffle(C0, C1, 0xD8);
+                C2 = Sse.Shuffle(C2, C3, 0xD8);
+                C4 = Sse.Shuffle(C4, C5, 0xD8);
+                C6 = Sse.Shuffle(C6, C7, 0xD8);
+
+                C0 = Sse.Shuffle(C0, C0, 0xD8);
+                C2 = Sse.Shuffle(C2, C2, 0xD8);
+                C4 = Sse.Shuffle(C4, C4, 0xD8);
+                C6 = Sse.Shuffle(C6, C6, 0xD8);
+
+                // Get the determinant
+                vTemp2 = row1;
+                Vector128<float> vTemp = Sse.Multiply(C0, vTemp2);
+                vTemp2 = Sse.Shuffle(vTemp2, vTemp, 0x40);
+                vTemp2 = Sse.Add(vTemp2, vTemp);
+                vTemp = Sse.Shuffle(vTemp, vTemp2, 0x30);
+                vTemp = Sse.Add(vTemp, vTemp2);
+                vTemp = Sse.Shuffle(vTemp, vTemp, 0xAA);
+
+                // Check determinant is not zero
+                if (MathF.Abs(vTemp.GetElement<float>(0)) < float.Epsilon)
+                {
+                    result = new Matrix4x4(float.NaN, float.NaN, float.NaN, float.NaN,
+                                float.NaN, float.NaN, float.NaN, float.NaN,
+                                float.NaN, float.NaN, float.NaN, float.NaN,
+                                float.NaN, float.NaN, float.NaN, float.NaN);
+                    return false;
+                }
+
+                Vector128<float> ones = Vector128.Create(1.0f, 1.0f, 1.0f, 1.0f);
```

Comment:
nit: It's better to use `Vector128.Create(1.0f)`

Reply:
Updated that line and the new test case I added.